dspy/clients/lm.py

```diff
@@ -60,9 +87,76 @@ def __call__(self, prompt=None, messages=None, **kwargs):
         
         return outputs
 
+    def copy(self, **kwargs):
+        """Create a copy of the current LM with updated kwargs."""
+        kwargs = {
+            "model": self.model,
+            "model_type": self.model_type,
+            "cache": self.cache,
+            "launch_kwargs": self.launch_kwargs,
+            **self.kwargs,
+            **kwargs,
+        }
+        return self.__class__(**kwargs)
+    
     def inspect_history(self, n: int = 1):
         _inspect_history(self, n)
 
+    def launch(self):
+        """Send a request to the provider to launch the model, if needed."""
```

Comment:
Bad design here. We should be getting a Provider object from the user, for example, and just calling `provider.launch()` or something like that. If statements per provider will not scale.

Reply:
(In next PR)